### ISSUE TITLE
CHParsed, ACParsed, and SplitCHParsed need to be renamed

### ISSUE THREAD

**Issue by dschwen** [idaholab/moose#5600]
These should be renamed to ```CahnHilliard```, ```AllenCahn```, and ```SplitCahnHilliard```. The old names are cryptic and inaccurate. The point of these kernes is not to parse anything. These kernels are the standard way of doing phase field now. This should be reflected by the more general name.

**Comment by permcody** [comment]
:+1:

**Comment by friedmud** [comment]
So they don't take any parsed functions as parameters?

**Comment by dschwen** [comment]
No, they take material properties as input parameters for mobility and free energy. Those material properties should have corresponding derivatives (through the DerivativeMaterialInterface). But they do not have to be provided by any parsed material. We can have and do have hardcoded free energy materials (with derivatives) as well.
Parsed functions (as in MOOSE Functions) do not play a role here (we need to couple to non-linear variables, which limits us to materials, as those are guaranteed to be evaluated at QPs only, as opposed to MOOSE functions which can be evaluated anywhere in space).

**Comment by dschwen** [comment]
I'm having second thoughts about the proposed names. The kernels implement only components of the Cahn-Hilliard and Allen-Cahn equations. ```CHParsed```/```CahnHilliard``` needs to be combined with ```CHInterface``` for example (```ACParsed``` and ```ACInterface``` are the same and ```SplitCHParsed``` needs ```SpitCHWRes```).
So it might be better to keep a homogeneous naming scheme and go with
```
CHParsed -> CHFreeEnergy
ACParsed -> ACFreeEnergy
SplitCHParsed -> SplitCHFreeEnergy
```
instead... :-/

**Comment by permcody** [comment]
@dschwen - old ticket, probably should be closed?

**Comment by dschwen** [comment]
It is not done. CHParsed and ACParsed are renamed, but SplitCHParsed still
has its old name.
